tensorflow/core/platform/default/device_tracer.cc

```diff
@@ -152,42 +152,14 @@ class CUPTIManager {
   // Disable tracing.  No further events will be delivered to 'client'.
   Status DisableTrace(CUPTIClient *client);
 
-  void MapThreadToClient(std::thread::id tid, CUPTIClient* client) {
+  void MapCidToClient(uint32 correlation_id, CUPTIClient* client) {
```

Comment:
It maps the correlation id to the same client all the times. If we know a tid is mapped to a client at a time, we don't need to change the mapping mechanism. Instead, we can remove `// Mapping tid -> client more than twice 	
    if (thread_to_client_.find(tid) != thread_to_client_.end()) {	
      CHECK(thread_to_client_[tid] == client);	
      return;	
    }`